### DISCUSSION TITLE
MMpose .pth models to .onnx

### DISCUSSION THREAD

**Discussion by Monalsingh** [open-mmlab/mmdeploy#668] [Q&A]
Hi there,

I am trying to convert the top-down pose model (mobilenetv2 & hrnet) present in MMpose to ONNX format. I was not able to convert it as its saying mmdeploy - ERROR - 'TopDown is not in the model's registry'. Is there a way to do so?

Command used:

`python ${MMDEPLOY_DIR}/tools/torch2onnx.py \
    ${MMDEPLOY_DIR}/configs/mmdet/detection/detection_tensorrt_dynamic-320x320-1344x1344.py \
    /home/monal/mmpose/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py \
    ${CHECKPOINT_DIR}/hrnet_w32_coco_256x192-c78dce93_20200708.pth \
    ${MMDET_DIR}/demo/demo.jpg \
    ${WORK_DIR}/mobilenetv2.onnx`

Output:

> 2022-06-28 10:49:07,155 - mmdeploy - INFO - torch2onnx:
>         model_cfg: /home/monal/mmpose/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py
>         deploy_cfg: /home/monal/mmdeploy/configs/mmdet/detection/detection_tensorrt_dynamic-320x320-1344x1344.py
> 2022-06-28 10:49:08,295 - mmdeploy - ERROR - 'TopDown is not in the models registry'
> 2022-06-28 10:49:08,295 - mmdeploy - ERROR - torch2onnx failed.
>

**Comment by RunningLeon** [comment]
@Monalsingh Hi, sorry for the late reply. for mmpose models, you should use the right deploy config: [configs/mmpose/pose-detection_tensorrt_static-256x192.py](https://github.com/open-mmlab/mmdeploy/blob/master/configs/mmpose/pose-detection_tensorrt_static-256x192.py)
